(DPS-3084) Revamp Management Overview (#981)
livechat/livechat-public-docs: content/management/index.mdx

## API versioning

The LiveChat Configuration API adopts the following versioning convention:

- **stable**

  The latest stable version we encourage developers to use. You can only expect bug fixes; no new new functionalities or breaking changes will be introduced.

- **developer preview (dev preview)**

  The unstable version that provides a preview of the upcoming functionalities. **It's not open for public use by default.** If you want to use it, contact us at developers@livechat.com or <a href="#open-chat" onClick={openChatWindow}>ask on the chat</a>, and we'll give you access.

- **legacy**

  The old version we still support. We highly encourage you to switch to the latest stable version.

**Currently:**

|                 | Configuration API |
| --------------- | ----------------- |
| **stable**      | 3.3               |
| **dev preview** | 3.4               |
| **legacy**      | 2.0, 3.1, 3.2     |

<Warning>

Currently, Configuration API v2 and v3 cover different resources. However, if you're still using v2, we encourage you to switch to v3 as more and more resources are being migrated to v3.

</Warning>

> # Resources

Author: I'm not sure about this headline as we tend to use "data structures" instead of "resources". But "data structures" seems flat to me... As if it was just a JSON. WDYT?

Reviewer: I like "resources" here, when we write "data structures" we refer only to the shape of some objects, not the resource itself.

Reviewer: I like "resources" here, when we write "data structures" we refer only to the shape of some objects, not the resource itself.

## Agents

> An agent is a type of user who communicates with customers. You can see the agent data structure [here](/messaging/agent-chat-api/#agent). The Configuration API allows for multiple agent-related actions, including creating, updating, deleting, and suspending agents.

Reviewer: >You can see the agent data structure...
...how it looks as a user in a chat.

Anyway, I'm not sure if linking to the agent **in chat** is ok, because it's configuration-api and agent data structure here looks a bit different (see `get-agent` method).

Author: We can link to Get Agents 👍🏻 And we can also change it here: https://developers.livechat.com/docs/management/configuration-api/#agents

<SectionLink to={"/management/configuration-api/#agents"}>
  See agent methods
</SectionLink>

## Auto access

> Auto access is a functionality that allows for assigning customers to dedicated groups based on the URL or the customer's geolocation.

Reviewer: >to dedicated groups based on
based on different rules like a URL of the website or the customer's geolocation.

In fact, there are more rules to set, but we don't need to mention all of them here.

How about describing full context here? Because a developer has no idea what is customer group and why they would want to set it (automatically).

>...When such customer starts a chat, the chat will be routed to the specific group.

Author: That might be a good idea

<SectionLink to={"/management/configuration-api/#auto-access"}>
  See auto access methods
</SectionLink>

## Bots

Bots are similar to regular agents. The main difference between them is that bots have a greater potental for automation, for example, they can listen for and react to incoming webhooks and pushes.

Bots are created and managed via the Configuration API. Just as regualar agents, bots can call Agent Chat API by the Web or RTM API.

> Bots are authorized with the use of the [agent token](/authorization/authorizing-api-calls/#agent-authorization-flows). Using bots requires sending the `X-Author-Id` header for the Web API and the `author_id` property for the RTM API.

Reviewer: Perhaps, we should mention here that bot status can be set by using agent-chat-api and `set_routing_status` method. It can be important here.

Author: We can do that 👍🏻

<SectionLink to={"/management/configuration-api/#groups"}>
  See group methods
</SectionLink>

## Properties

Properties are key-value storages. Depending on the API version, they can be set within the following locations: a chat, a thread, an event, a group, and a license. Property configuration is owned by a Client Id (integration), not by license (v3.3+). Properties can be public or private.

<SectionLink to={"/management/configuration-api/#properties"}>
  Learn more about properties
</SectionLink>

<SectionLink to={"/management/configuration-api/#methods-4"}>
  See property methods
</SectionLink>

## Webhooks

In a nutshell, actions invoked by the use of Messaging APIs or the Configuration API result in events. You can be notified about those events with webhooks. LiveChat provides a number of webhooks, which you can register and manage via the Configuration API. Just like properties, webhooks are registered per Client Id (integration), not per license (v3.3+).

We can distinguish two types of webhooks:

- license webhooks
- bot webhooks

> Once **license webhooks** are set up, you will always receive them. Read our tutorial on <a href="https://developers.livechat.com/updates/setting-up-webhooks-with-livechat-api/" target="_blank">how to configure LiveChat license webhooks via the Configuration API</a>.

Reviewer: by setting up you meain registering and **enabling** them, right?

Author: Yeah :) I think it's okay for a general overview because a developer might want to "do something with properties, some configuration/setup), and doesn't really know that props need to be registered & enabled.

Reviewer: ok

# Introduction

The Configuration API allows you to manage resources such as **agents, auto access, bots, groups, properties, and webhooks**. You can use it to build advanced integrations as well as alter the configuration of your resources directly via the API.

## Use cases

Here are some of the most common use cases of the Configuration API:

> - Build integrations that connect LiveChat with other services via **webhooks**. React to events happening in the chat.

Reviewer: you're right

- **legacy**

  The old version we still support. We highly encourage you to switch to the latest stable version.

**Currently:**

|                 | Configuration API |
| --------------- | ----------------- |
| **stable**      | 3.3               |
| **dev preview** | 3.4               |
| **legacy**      | 2.0, 3.1, 3.2     |

<Warning>

Currently, Configuration API v2 and v3 cover different resources. However, if you're still using v2, we encourage you to switch to v3 as more and more resources are being migrated to v3.

</Warning>

# Resources

## Agents

An agent is a type of user who communicates with customers. You can see the sample agent data structure in the [response of Get Agent](/management/configuration-api/#get-agent). The Configuration API allows for multiple agent-related actions, including creating, updating, deleting, and suspending agents.

<SectionLink to={"/management/configuration-api/#agents"}>
  See agent methods
</SectionLink>

## Auto access

> Auto access is a functionality that allows for assigning customers to dedicated groups based on the URL or the customer's geolocation. When a customer who matches some pre-defined auto access rules starts a chat, the chat is automatically routed to the specific group. You can use auto access, for example, to provide multi-language support with each group of agents being responsible for communication in a different language.

Reviewer 2: ```suggestion
Auto access is a functionality that allows for assigning customers to dedicated groups based on the visited URL or the customer's geolocation. When a customer who matches some pre-defined auto access rules starts a chat, the chat is automatically routed to the specific group. You can use auto access, for example, to provide multi-language support with each group of agents being responsible for communication in a different language.
```

<SectionLink to={"/management/configuration-api/#auto-access"}>
  See auto access methods
</SectionLink>

## Bots

Bots are similar to regular agents. The main difference between them is that bots have a greater potental for automation, for example, they can listen for and react to incoming webhooks and pushes.

> Bots are created and managed via the Configuration API. Just as regualar agents, bots can call Agent Chat API by the Web or RTM API.

Reviewer 2: ```suggestion
Bots are created and managed via the Configuration API. Just as regular agents, bots can call the Agent Chat API by either Web or RTM API.
```

Bots are authorized with the use of the [agent token](/authorization/authorizing-api-calls/#agent-authorization-flows). Using bots requires sending the `X-Author-Id` header for the Web API and the `author_id` property for the RTM API.

To change a bot's status (`accepting_chats`, `not_accepting_chats`, or `offline`), use the [Set Routing Status](/messaging/agent-chat-api/#set-routing-status) method from the Agent Chat API.

<SectionLink to={"s/management/configuration-api/#bots"}>
  See bot methods
</SectionLink>

## Groups

> Groups let you organize your work by creating teams with agent and bot members. You can use groups to do configuration based on shared settings (language, working hours, department) and to separate chat routing.

Reviewer 2: This one seems a little short compared to the other ones, maybe we can add a note similar to the one we added for the agents?

```suggestion
Groups let you organize your work by creating teams with agent and bot members. You can use groups to do configuration based on shared settings (language, working hours, department) and to separate chat routing. With the Configuration API, you can perform actions such as creating and deleting a group, or updating its' settings.
```
